platform/wayland/cog-view-wl.c

```diff
@@ -393,16 +406,15 @@ validate_exported_geometry(uint32_t width, uint32_t height)
 static void
 on_export_shm_buffer(void *data, struct wpe_fdo_shm_exported_buffer *exported_buffer)
 {
-    CogWlPlatform *platform = (CogWlPlatform *) cog_platform_get();
-    CogWlViewport *viewport = COG_WL_VIEWPORT(platform->viewport);
-    CogWlView     *view = data;
+    CogWlView               *view = data;
+    g_autoptr(CogWlViewport) viewport = COG_WL_VIEWPORT(cog_view_get_viewport((CogView *) view));
 
     struct wl_resource   *exported_resource = wpe_fdo_shm_exported_buffer_get_resource(exported_buffer);
     struct wl_shm_buffer *exported_shm_buffer = wpe_fdo_shm_exported_buffer_get_shm_buffer(exported_buffer);
 
     uint32_t image_width = wl_shm_buffer_get_width(exported_shm_buffer);
     uint32_t image_height = wl_shm_buffer_get_height(exported_shm_buffer);
-    if (!validate_exported_geometry(image_width, image_height)) {
+    if (!viewport || !validate_exported_geometry(viewport, image_width, image_height)) {
```

Comment:
Still we need to dispatch and release the generated frame

Reply:
That makes sense, yes 🆗